File: 02 Jewels and Stones/Jewels and Stones.cpp
class Solution
{
public:
    int numJewelsInStones(string J, string S)
    {
        map<char, int> mp;
        for (int i = 0; i < S.size(); i++)
            mp[S[i]]++;
        int cnt = 0;
        for (int i = 0; i < J.size(); i++)
            cnt += mp[J[i]];
        return cnt;
    }
};

File: 02 Jewels and Stones/Jewels and Stones.js
/**
 * @param {string} J
 * @param {string} S
 * @return {number}
 */
var numJewelsInStones = function (J, S) {
    let mp = new Map();
    for (let i = 0; i < S.length; i++) {
        let x = 1;
        if (mp.has(S[i]))
            x += mp.get(S[i]);
        mp.set(S[i], x);
    }
    let cnt = 0;
    for (let i = 0; i < J.length; i++) {
        if (mp.has(J[i]))
            cnt += mp.get(J[i]);
    }
    return cnt;
};